src/Build/Definition/Project.cs

```diff
@@ -3441,7 +3441,7 @@ public bool IsSuitableExistingItemXml(ProjectItemElement candidateExistingItemXm
                     return false;
                 }
 
-                if ((metadata?.Any() == true) || candidateExistingItemXml.Count > 0)
+                if ((metadata != null && metadata.Any()) || candidateExistingItemXml.Count > 0)
```

Comment:
These ones don't need to be changed, they would return `false` if `metadata` is null. The only ambiguity is with `x?.Length == 0` where you (usually) want to check if it's `null` or has no elements.

Reply:
done